tests/components/controlBar-test.js

```diff
@@ -1624,4 +1650,79 @@ describe('ControlBar', function () {
+        { "name": "fullscreen", "location": "controlBar", "whenDoesNotFit": "keep", "minWidth": 45 },
+      ];
+      // Mock addEventListener on document object since TestUtils.Simulate will not work in this case
+      eventMap = {};
+      document.addEventListener = jest.genMockFn().mockImpl(function(event, cb) {
+        eventMap[event] = cb;
+      });
+      ReactDOM.render(
+        <ControlBar
+          {...baseMockProps}
+          controlBarVisible={true}
+          componentWidth={500}
+          playerState={CONSTANTS.STATE.PLAYING}
+          isLiveStream={baseMockProps.isLiveStream} />
+      , document.body);
+      ctrlBarElement = document.body.querySelector('.oo-control-bar');
+      focusableElements = ctrlBarElement.querySelectorAll('[data-focus-id]');
+      mockEvent = { key: CONSTANTS.KEY_VALUES.TAB, preventDefault: function() {} };
+    });
+
+    afterEach(function() {
+      ReactDOM.unmountComponentAtNode(document.body);
+      document.body.innerHTML = '';
+    });
+
+    it('should constrain tab navigation to control bar elements when in fullscreen mode', function() {
+      baseMockController.state.fullscreen = true;
+      // Tab on document, focuses first element
+      document.activeElement = null;
+      mockEvent.target = document.body;
+      eventMap.keydown(mockEvent);
+      expect(document.activeElement.getAttribute('data-focus-id')).toBe(focusableElements[0].getAttribute('data-focus-id'));
+      // Tab on last element, focuses on first
+      document.activeElement = null;
+      mockEvent.target = focusableElements[focusableElements.length - 1];
+      eventMap.keydown(mockEvent);
+      expect(document.activeElement.getAttribute('data-focus-id')).toBe(focusableElements[0].getAttribute('data-focus-id'));
+      // Shift + tab on document, focuses on last element
+      document.activeElement = null;
+      mockEvent.target = document.body;
+      mockEvent.shiftKey = true;
+      eventMap.keydown(mockEvent);
+      expect(document.activeElement.getAttribute('data-focus-id')).toBe(focusableElements[focusableElements.length - 1].getAttribute('data-focus-id'));
+      // Shift + tab on first element, focuses on last
+      document.activeElement = null;
+      mockEvent.target = focusableElements[0];
+      mockEvent.shiftKey = true;
+      eventMap.keydown(mockEvent);
+      expect(document.activeElement.getAttribute('data-focus-id')).toBe(focusableElements[focusableElements.length - 1].getAttribute('data-focus-id'));
+    });
+
+    it('should NOT constrain tab navigation to control bar elements when NOT in fullscreen mode', function() {
+      baseMockController.state.fullscreen = false;
+      // Tab on last focusable element, should NOT go back to the first
+      document.activeElement = null;
+      mockEvent.target = focusableElements[focusableElements.length - 1];
+      eventMap.keydown(mockEvent);
+      expect(document.activeElement).toBeNull();
```

Comment:
Is there a better way to test that we are not constraining tab navigation to control bar elements? Can we add non-player elements to our test DOM and check if those become the active element?

Reply:
I actually tried this first but I think jsdom only goes as far as implementing the `focus()` function on elements, but it will not react to tab key events. This behavior would have to be mocked, so I don't think it helps a lot

Reply:
Have you seen any issues with manual testing? This sounds fine.

Reply:
Nope, I tested this and it looks ok